apis/r/R/Init.R

```diff
@@ -5,6 +5,9 @@
 .onLoad <- function(libname, pkgname) {
     rpkg_lib_version <- tiledb::tiledb_version(compact=TRUE)
     soma_lib_version <- libtiledbsoma_version(compact=TRUE)
+    # Check major and minor but not micro: sc-50464
+    rpkg_lib_version <- strsplit(as.character(rpkg_lib_version), "\\.")[[1]][1:2]
+    soma_lib_version <- strsplit(soma_lib_version, "\\.")[[1]][1:2]
     if (rpkg_lib_version != soma_lib_version) {
         msg <- sprintf("TileDB Core version %s used by TileDB-R package, but TileDB-SOMA uses %s",
```

Comment:
At this point, `rpkg_lib_version` and `soma_lib_version` will be vectors, not scalars, and `if` is **not** vectorized in R; both of these vectors will need to be collapsed into a single character value for proper comparisons

```R
> rpkg_lib_version <- tiledb::tiledb_version(compact=TRUE)
> rpkg_lib_version <- strsplit(as.character(rpkg_lib_version), "\\.")[[1]][1:2]
> rpkg_lib_version
[1] "2"  "24"
> paste(rpkg_lib_version, collapse = ".") # collapse into a version string
[1] "2.24"
```